morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -463,13 +399,30 @@ public void ensureIndexes(final boolean background) {
         return getByKeys(null, keys);
     }
 
+    private DBCollection getCollection(final Object obj) {
+        if (obj == null) {
+            return null;
```

Comment:
I get that.  But as a private method, is there ever a case where the caller would be expected to sometimes pass in null and then check for a null response?  If not, better to just let this method throw rather than return null.   The caller could just as easily not call the method if the argument is null.